lambda/simplif.ml

```diff
@@ -383,15 +402,17 @@ let simplify_lets lam =
   | Lconst _ -> ()
   | Lvar v ->
       use_var bv v 1
-  | Lapply{ap_func = Lfunction{kind = Curried; params; body}; ap_args = args}
-    when optimize && List.length params = List.length args ->
-      count bv (beta_reduce params body args)
-  | Lapply{ap_func = Lfunction{kind = Tupled; params; body};
-           ap_args = [Lprim(Pmakeblock _, args, _)]}
-    when optimize && List.length params = List.length args ->
-      count bv (beta_reduce params body args)
-  | Lapply{ap_func = l1; ap_args = ll} ->
-      count bv l1; List.iter (count bv) ll
+  | Lapply{ap_func = ll; ap_args = args} ->
+      let no_opt () = count bv ll; List.iter (count bv) args in
+      begin match ll with
+      | Lfunction lf ->
```

Comment:
You dropped the `optimize` condition.  Is that on purpose?

Reply:
This was a mistake, which is now fixed. Thanks!